### DISCUSSION TITLE
Help with improving R2 using autoML

### DISCUSSION THREAD

**Discussion by sgujjaABM** [pycaret/pycaret#1400] [Q&A]
Hi, I am implementing pycaret autoML function for regression analysis, and following the sample code in the link https://pycaret.org/automl/ for my dataset. Predicting on the test data set results in R2 of 0.57, while R2 from CV is 0.55. Can you please suggest any options for hyperparameter tuning to further optimize R2? Thanks for helping.
**Describe the bug**

**To Reproduce**

```python
regr = setup(dataset, target = 'y',numeric_features =features,normalize=True,n_jobs=1) 

# compare all baseline models and select top 5
top5 = compare_models(n_select = 5)
# tune top 5 base models
tuned_top5 = [tune_model(i) for i in top5]
# ensemble top 5 tuned models
bagged_top5 = [ensemble_model(i) for i in tuned_top5]
# blend top 5 base models
blender = blend_models(estimator_list = top5)
# select best model
best = automl(optimize = 'R2')
train_metrics = pull()
print(train_metrics)
save_model(best, 'best-model')

# Deploy Model to generate predictions
test_predictions = predict_model(best)
test_metrics = pull()
print(test_metrics)
print(test_predictions.head(5))
```

**Expected behavior**

**Additional context**

**Versions**
<details>

</details>

**Comment by ngupta23** [comment]
You could try some feature interactions to see if it improves performance. Pass these arguments to the setup. 

```python
polynomial_features = True, polynomial_degree = 2
```

Apart from this, you may have to apply some domain specific knowledge and create manual features with good predictive power before feeding to pycaret.
